### DISCUSSION TITLE
Importing Sony Footage: FCPX Assumes Full or Video Range?

### DISCUSSION THREAD

**Discussion by skymakai** [CommandPost/FCPCafe#263] [Final Cut Pro Questions]
I've been looking everywhere to try and find an answer for this question.  I'm hoping the experts here might be able to advise.

Working with footage from various cameras, some are much more simple than others. Canon and DJI footage, for example, is simple.  It's recorded in video range, viewed in video range, exported in video range.

Sony S-Log3 footage, though, is not so simple.  It's recorded in full range.  
MXF Sony S-Log3 footage is recognized by FCPX, and assumed full range, and an S-Log3 input conversion LUT is automatically applied.  But MP4 Sony S-Log3, FCPX doesn't recognize that as S-Log3 footage.

Without the ability to tell FCPX whether files are full or video range (like Resolve), how do we properly handle full range video?

**Comment by d-mac-648** [comment]
First of all, you can, and probably should, turn off the auto-applied LUT for log footage. With FCP, any LUT applied in the inspector will put the footage into the timeline's color space (often Rec. 709). This can result in clipping of the original footage and thereby limiting the effect of any color work done. It's usually better to apply a LUT via the Custom LUT effect and place that last in the list of effects applied to a given clip in a timeline. That way, the color processing can take full advantage of the larger color space often provided with log footage. It's (LUT applied last) like using Scene referred processing versus display referred processing in Resolve, which can preserve more color info during grading. This issue is exacerbated when footage was recorded at less than the ideal exposure (and other settings in the camera) specified by a given LUT.

You can use something like MediaInfo or Invisor to check the levels range prior to import into FCP. Or, you can check the Sony specs. to see what the recorded range should be. Workflow testing is always a good idea. If you're just concerned with the MP4 footage not having a LUT auto-applied, see the above paragraph.

It should be noted that a technical/transform LUT is designed for a specific exposure (and other settings). So, applying one of the LUTs provided by Apple may not be the best fit, depending on the camera settings during filming.

FCP isn't particularly transparent about these things, nor does the user have as much control as one does in Resolve, for example.

I hope this makes sense and helps...

**Reply by skymakai** [reply]
Thank you for your reply.  I'm aware of the potential issues by keeping the automatically applied LUT in place, versus using a Custom LUT effect, in order to have the ability to correct for exposure, WB, etc. before the LUT is applied.  I'm also using Invisor as well, and know which footage is Full and which is Video range.

I guess I should have been more clear in my question, as I would like an expert to tell me... what range does FCPX assume?

If FXPX can read metadata from some files (MXF) and not from others, does FCPX assume Video range unless it sees Full range metadata?

**Reply by d-mac-648** [reply]
Sorry if my initial response was at all unclear.

I'm not sure anyone could be considered an expert with regards to this topic, as Apple doesn't provide any really useful info about the process of how FCP interprets colorspace/color profile info on import. Apple is very opaque about many things regarding FCP, likely due in part to some misguided effort to make things easier for users but that in reality just muddles things more.

I would imagine that FCP looks in a file on import to try to find info about things like levels, either from metadata or file headers. The only place I know of where a user can change the interpretation of the color space of imported clips is via the Color Space Override menu in the Settings "panel" of the Inspector. Though, the choices aren't very fine-grained.

Other than that, I would suggest doing some testing as I mentioned in my earlier post. Record some test footage, noting the recording settings, and include color bars. Use the scopes in FCP to see how the file has been interpreted. I also use ScopeBox when doing levels (and other) testing. I would hesitate to even write an article about how FCP handles such things, as the current correct answer, at any given point in time, is a moving target.

I would say that this approach is superior to just saying FCP assumes this or that, as recording settings in cameras may not be correct or the behavior of FCP and macOS may change after any update. Firmware changes to cameras or devices like the Atomos Ninja V can affect how video is interpreted and recorded. Some "non-professional" file formats like MP4 files, or certain container formats may have varying support for proper metadata (and things can always be improperly implemented). 

So, my best (and perhaps not "easy button" satisfying) answer is to test a workflow to verify things and then don't change FCP, macOS, camera firmware, or recording device versions (or anything else in your workflow chain) without testing things again.

I've found this article to be especially informative and helpful in trying to better understand this issue (though, you may have already seen it):

https://www.thepostprocess.com/2019/09/24/how-to-deal-with-levels-full-vs-video/


Cheers.

**Reply by joema4** [reply]
S-Log3 is by definition data levels, so any NLE should interpret and display both image and waveform accordingly. E.g, if you expose a 90% reflective white target at 61% IRE (which is Sony's recommended S-Log3 level), the NLE without any conversion should display 61% on the luma waveform. See the attached table for Sony's recommended code values and IRE levels.

I have tested this in various MP4 formats from the A7SIII and also XAVC-I MXF from the FX6. Attached is one test using Resolve, but I've done the same thing with FCP. If those cameras are white balanced and exposed properly, the color is almost identical. The luma waveforms shown are post conversion, so those are Rec.709 levels. The pre-conversion S-Log3 level for white was 61% on the waveform, in camera and after import by the NLE.

FCP (and Resolve) do not recognize the color metadata in the Sony MP4 container, so they won't automatically apply any conversion. The MXF container has much richer metadata, so both NLEs can apply automatic conversion.

FCP also cannot read clip timecode from Sony's MP4 formats, but Resolve can. There is a 3rd-party utility QTChange that can rewrite the Sony MP4 clips so the timecode is readable by FCP: https://www.videotoolshed.com/handcrafted-timecode-tools/qtchange/

If recording ProRes on an Atomos, there is a setting for "legal levels". You don't want that.

![SLog3_SLog3_IRE_10Bit_CodeValues_Black_Gray_White](https://github.com/CommandPost/FCPCafe/assets/65232258/89a4eb37-9e62-4489-8d3f-c82ff1fedc84)
![Dual_A7S3_FX6_WB_Test](https://github.com/CommandPost/FCPCafe/assets/65232258/79399e10-f8a1-4cfd-95f4-24a32a8c0b84)
